src/sptps_test.c

```diff
@@ -125,6 +134,149 @@ static void usage() {
 	fprintf(stderr, "Report bugs to tinc@tinc-vpn.org.\n");
 }
 
+#ifdef HAVE_MINGW
+
+// Windows does not allow calling select() on anything but sockets. Therefore,
```

Comment:
The proper way on Windows is to use `WSAWaitForMultipleEvents()`, and there is some way to have that work with BSD sockets. This is what's done in `event.c`. It might actually be less code to go that way than to have a separate thread and TCP socket. But we can also add that later.